lib/pngdriver/read_ppm.c

```diff
@@ -36,7 +37,8 @@ void read_ppm(void)
     if (fscanf(input, "P6 %d %d %d", &i_width, &i_height, &maxval) != 3)
         G_fatal_error("PNG: invalid input file %s", png.file_name);
 
-    fgetc(input);
+    if (fgetc(input) == EOF)
+        G_fatal_error(_("PNG: unexpected EOF reading %s"), png.file_name);
```

Comment:
This message could be a bit cryptic to a user, let us use "PNG: invalid input file %s".